src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -548,6 +554,30 @@ class RsErrorAnnotator : AnnotatorBase(), HighlightRangeExtension {
         }
     }
 
+    private fun checkSuperTraitImplemented(holder: RsAnnotationHolder, impl: RsImplItem, trait: RsTraitItem) {
+        if (impl.project.macroExpansionManager.macroExpansionMode !is MacroExpansionMode.New) return
+
+        val traitRef = impl.traitRef ?: return
+        val typeRef = impl.typeReference ?: return
+        val type = typeRef.type
+        val supertraits = trait.typeParamBounds?.polyboundList?.mapNotNull { it.bound } ?: return
+        val lookup = typeRef.implLookup
+
+        val substitution = impl.implementedTrait?.subst ?: emptySubstitution
```

Comment:
I found one more false-positive:
```rust
trait Tr1<A=Self> {}
trait Tr2<A=Self> : Tr1<A> {}

struct Sss;
impl Tr1 for Sss {}
impl Tr2 for Sss {}
```

Should be fixed by this:
```suggestion
        val selfSubst = mapOf(TyTypeParameter.self() to type).toTypeSubst()
        val substitution = (impl.implementedTrait?.subst ?: emptySubstitution).substituteInValues(selfSubst) + selfSubst
```

Reply:
Thank you!